app/src/main/java/com/example/androidtrainingproject/ui/products/ProductDetailsScreen.kt

```diff
@@ -17,18 +17,25 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
 import com.example.androidtrainingproject.R
 import com.example.androidtrainingproject.ui.shared.AppTopBar
 import com.example.androidtrainingproject.ui.shared.WideButton
 import com.example.androidtrainingproject.ui.theme.LabelText
 
 @Composable
-fun ProductDetails() {
+fun ProductDetails(navController: NavController, productId: Int) {
     val productDetailsViewModel: ProductDetailsViewModel = hiltViewModel()
+    productDetailsViewModel.getProductById(productId, "*")
```

Comment:
Such invocations shouldn't happen out in the open as they will trigger on each minor rerender.

Best option: When you have implemented the new navigation library use the `SavedStateHandle` demonstrated in the last lecture to provide the navigation arguments directly to the view model so it can handle this logic itself on init.

Alternative until the new navigation library is used: Place this logic inside of a `LaunchedEffect(Unit)` so that it triggers only once.